types/transaction.go

```diff
@@ -14,9 +14,9 @@ type TransactionData struct {
 }
 
 // NewTransactionData builds a new TransactionData instance
-func NewTransactionData(msg sdk.Msg, msgs ...sdk.Msg) *TransactionData {
+func NewTransactionData(msgs ...sdk.Msg) *TransactionData {
```

Comment:
Althought this makes sense, I have two questions: 

*1.* Is this backward compatible?  
If someone was using the library as 
```go
data := NewTransactionData(firstMessage, otherMessages...)
```
Does this still work? 

*2.*  This code allows to not pass any `msgs` while building the data itself. I think this case should be checked when building the transaction to be broadcast, in order to avoid any error: building a transaction with no messages does not make any sense at all.

Reply:
> 1. Is this backward compatible?
It is partially backward compatible, the order of msgs will be changed. For instance, `NewTransactionData(1, 2, 3)` returns `2, 3, 1` order while new `NewTransactionData(1, 2, 3)` will returns `1, 2, 3`,  see:
https://go.dev/play/p/xrpQfAwIsf3

> 2. This code allows to not pass any msgs while building the data itself. 
Yeah, I agree with you, it the same we put `nil` into current `NewTransactionData`, we should check the msgs is not empty when building transaction.

Reply:
Added the check for the case `building a transaction with no messages`